dashboard/app/controllers/sections_controller.rb

```diff
@@ -1,5 +1,17 @@
 class SectionsController < ApplicationController
-  before_action :load_section_by_code, except: :update
+  layout false, only: :index
+  before_action :load_section_by_code, only: [:log_in, :show]
+
+  # Get the set of sections owned by the current user
+  def index
+    if !current_user
+      head :forbidden
+    elsif current_user.teacher?
```

Comment:
I would have thought that if I'm a student, `current_user.sections` would return `[]` such that we could just combine the last two blocks?

We have a separate concept of `sections_as_student` (for sections I'm in vs. section I own).